cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetector.java

```diff
@@ -183,11 +197,43 @@ public void shutdown() {
     LOG.info("Anomaly detector shutdown completed.");
   }
 
-  public AnomalyDetectorState anomalyDetectorState() {
+  public synchronized AnomalyDetectorState anomalyDetectorState() {
+    // Retrieve mean time between anomalies
+    Map<AnomalyType, Double> meanTimeBetweenAnomalies = new HashMap<>(AnomalyType.cachedValues().size());
+    for (AnomalyType anomalyType : AnomalyType.cachedValues()) {
+      meanTimeBetweenAnomalies.put(anomalyType, _anomalyRateByType.get(anomalyType).getMeanRate());
+    }
+    // Retrieve mean time to start a fix
+    double meanTimeToStartFix = _ongoingAnomalyCount == 0L ? 0 : _ongoingAnomalyDurationSumForAverageMs / _ongoingAnomalyCount;
+
+    _anomalyDetectorState.setMetrics(new AnomalyMetrics(meanTimeBetweenAnomalies, meanTimeToStartFix, _numSelfHealingStarted));
     _anomalyDetectorState.setSelfHealingEnabledRatio(_anomalyNotifier.selfHealingEnabledRatio());
     return _anomalyDetectorState;
   }
 
+  /**
+   * If there is an ongoing anomaly, clear the earliest detection time to indicate start of an ongoing fix.
+   */
+  public synchronized void maybeClearOngoingAnomalyDetectionTimeMs() {
+    if (_ongoingAnomalyDetectionTimeMs != NO_ONGOING_ANOMALY_FLAG) {
+      double elapsed = _time.milliseconds() - _ongoingAnomalyDetectionTimeMs;
+      _ongoingAnomalyDurationSumForAverageMs += elapsed;
```

Comment:
IIUC, `_ongoingAnomalyDurationSumForAverageMs` and `_ongoingAnomalyCount` are used to calculate `meanTimeToStartFix`(though a division),   but their value are updated in two places. `_ongoingAnomalyCount` is updated when an anomaly is detected; `_ongoingAnomalyDurationSumForAverageMs` is updated when any execution starts.

There are two scenario I think that may generate confusing `meanTimeToStartFix` metrics.
(1)  t0: a goal violation is detected 
       t1 : self-healing disabled/failed 
      t2 : user send a request to increase RF of a topic 
      in this case, looks like CC will take RF increasing operation as a self-healing operation, and `meanTimeToStartFix` will be `t2 - t0`

(2) t0: a goal violation detected
     t1: self-healing succeed 
     t2: another goal violation detected 
     t3: self-healing failed
     in this case, `meanTimeToStartFix` will be `(t1 - t0)/2`, shouldn't it be `(t1 - t0)`?

Reply:
Thanks for the input!

The first scenario:
* Please see the JavaDoc for the `AnomalyMetrics` class, which defines `mean time to start fix` as follows:
```
Mean time between the detection of earliest ongoing anomaly and start of a proposal execution. This metric covers executions started both by anomaly detector and as a result of on-demand requests.
```
This means that any execution (e.g. a `rebalance`, `add`/`remove`/`demote` broker operation, `PLE`, or `RF increase` that start replica movements/addition/deletion or leadership transfer) is considered as an action towards a fix. If it is (1) a self-healing action, then the designated mitigation process is already well-defined a priori, (2) an on-demand request, then the cluster operator intends to perform an admin operation to address a certain issue -- e.g. cluster expansion, replacement of bad hardware, alleviating the load from selected set of brokers. Note that all such admin operations that start an execution use a list of goals to keep the cluster in the desired state (e.g. no offline replicas, satisfy goals). Regardless, even if the state after the fix still violates a goal or cluster operator does not know what s/he is doing, then this will trigger another anomaly and cause a rise in the mean time to start a fix.

The second scenario:
* Good point! It should indeed be `(t1 - t0)` (will fix it). That brings up the question of capturing "how long has the latest unfixed/unfixable anomaly there". I will also add a new metric to represent that value.